components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2951,51 +2956,75 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             throw new ResponseError(ErrorCodes.BAD_REQUEST, "invalid role name");
         }
 
-        await this.checkAndBlockUser("setTeamMemberRole");
-        await this.guardTeamOperation(teamId, "update", "org_members_write");
+        const requestor = await this.checkAndBlockUser("setTeamMemberRole");
+        const { team } = await this.guardTeamOperation(teamId, "update", "org_members_write");
 
-        await this.teamDB.setTeamMemberRole(userId, teamId, role);
+        const centralizedPermsEnabled = await this.centralizedPermissionsEnabled(requestor, team.id);
+        if (centralizedPermsEnabled) {
+            const writeRequest =
+                role === "owner" ? organizationOwnerRole(team.id, userId) : organizationMemberRole(team.id, userId);
```

Comment:
Having one function and passing in the role as an argument seems better

Reply:
We can add a helper which does exactly this line, we just haven't needed it in multiple places yet

Reply:
We are also not using the two functions in more cases I guess, so I'd rather have one still rather short function than three functions tbh